source/common/upstream/load_balancer_impl.cc

```diff
@@ -804,23 +804,23 @@ HostConstSharedPtr EdfLoadBalancerBase::chooseHostOnce(LoadBalancerContext* cont
   }
 }
 
-HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector& hosts_to_use,
-                                                                const HostsSource&) {
+HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector&,
+                                                                const HostsSource& source) {
   HostSharedPtr candidate_host = nullptr;
-  for (uint32_t choice_idx = 0; choice_idx < choice_count_; ++choice_idx) {
-    const int rand_idx = random_.random() % hosts_to_use.size();
-    HostSharedPtr sampled_host = hosts_to_use[rand_idx];
+  uint64_t candidate_active_rq = 0;
```

Comment:
Assuming we keep this implementation can you add more comments about what is going on in this function? It would help readability.